Added supporting context for job suspension functionality (#35516)
kubernetes/website: content/en/docs/concepts/workloads/controllers/job.md

reset to the current time. This means that the `.spec.activeDeadlineSeconds`
timer will be stopped and reset when a Job is suspended and resumed.

> Remember that suspending a Job will delete all Pods that are not Completed. When the Job is
> suspended, Non-Completed [Pods will be terminated](/docs/concepts/workloads/pods/pod-lifecycle/#pod-termination)

Reviewer: ```suggestion
When you suspend a Job, any running Pods that don't have a status of `Completed` will be [terminated](/docs/concepts/workloads/pods/pod-lifecycle/#pod-termination).
```

      ...
```

You can also toggle Job suspension by patching the Job using the command line.

Suspend an active Job:

```shell

> kubectl patch job/myjob -p '{"spec":{"suspend":true}}'

Reviewer 2: In docs, I like to write things longhand (it helps the reader);
```suggestion
kubectl patch job/myjob --type=strategic --patch '{"spec":{"suspend":true}}'
```

Now the reader can immediately see that this is a strategic merge patch.

Author: This is great @sftim - Thank you.  I've committed another change so both examples are in sync.

      ...
```

> Alternatively, a job suspension can be toggled from the command line by patching the job resource.  Suspend an active job:

Reviewer: ```suggestion
You can also toggle Job suspension by patching the Job using the command line.

Suspend an active job:
```

> Alternatively, a job suspension can be toggled from the command line by patching the job resource.  Suspend an active job:

Reviewer: I'm not sure how necessary the command-line steps are to add though. We don't currently give full instructions for updating the manifest.

Author: Thanks for your awesome suggestions above @shannonxtreme, I think this covers off this nicely 😊. Will make another commit following this discussion change one of the 'job' references to 'Job' for consistency.

As for the command line patch options, I'm happy to go either way on this.  The main objective on this PR from my side was to get the main message that we've already covered above.

I did however find myself looking for a convenient means of toggling the suspend functionality within the CLI, hence it's inclusion in the PR.

Personally, I use the CLI more often where there is a shortcut (akin to kubectl create) or, for reactive changes which arguably a job suspension/resume would be.  I do however understand your feedback and viewpoint regarding the docs providing instructions on updating manifests.

I'll leave it at your discretion with no regrets if it's removed from the PR, let me know and if needed I'll update the commit.

Kind Regards

James Spurin

```shell
kubectl patch job/myjob -p '{"spec":{"suspend":true}}'
```

> Unsuspend an active job:

Reviewer: ```suggestion
Resume a suspended Job:
```
